parasoft-findings-teamcity-agent/src/test/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcessTest.java

```diff
@@ -0,0 +1,226 @@
+            Mockito.verify(buildProgressLogger).message("Transforming " + testReport.getAbsolutePath() + " with Parasoft SOAtest");
+        } finally {
+            deleteIfExists(pmdReport);
+            deleteIfExists(junitReport);
+        }
+    }
+
+    @Test
+    public void test_transformUnitTestReport() throws Throwable {
+        File junitReport = new File(reportDirPath + "/junit-jtest_report.xml");
+        File testReport = new File(reportDirPath + "/jtest_report.xml");
+
+        try {
+            // Given
+            setupMockedDataForTransformation("reports/jtest_report.xml");
+
+            // When
+            BuildFinishedStatus status = parasoftFindingsBuildProcess.call();
+            int invalidReportCount = parasoftFindingsBuildProcess.getInvalidReportCount();
+
+            // Then
+            Assertions.assertTrue(junitReport.exists());
+            Assertions.assertEquals(status.name(), "FINISHED_SUCCESS");
+            Assertions.assertEquals(invalidReportCount, 0);
+
+            Mockito.verify(buildProgressLogger, Mockito.atLeastOnce()).message(Mockito.isA(String.class));
+            Mockito.verify(buildProgressLogger, Mockito.atLeastOnce()).logMessage(Mockito.isA(BuildMessage1.class));
+
+            Mockito.verify(buildProgressLogger).message("Transforming " + testReport.getAbsolutePath() + " with Parasoft Analyzers");
+            Mockito.verify(buildProgressLogger).message("Generated report with transformation: " + junitReport.getAbsolutePath());
+        } finally {
+            deleteIfExists(junitReport);
+        }
+    }
+
+    @Test
+    public void test_transform_reportNotFound() throws Throwable {
+        // Given
+        setupMockedDataForTransformation("report.xml");
+
+        // When
+        BuildFinishedStatus status = parasoftFindingsBuildProcess.call();
+
+        // Then
+        Assertions.assertEquals(status.name(), "FINISHED_SUCCESS");
+
+        Mockito.verify(buildProgressLogger, Mockito.atLeastOnce()).error(Mockito.isA(String.class));
+        Mockito.verify(buildProgressLogger).error("No reports found for pattern: report.xml");
+    }
+
+    @Test
+    public void test_transformFail_invalidContent() throws Throwable {
+        // Given
+        File testReport = new File(reportDirPath + "/invalidReport.xml");
+        File pmdReport = new File( reportDirPath + "/pmd-invalidReport.xml");
+        File pmdCpdReport = new File(reportDirPath + "/pmdCpd-invalidReport.xml");
+        setupMockedDataForTransformation("reports/invalidReport.xml");
+
+        // When
+        BuildFinishedStatus status = parasoftFindingsBuildProcess.call();
+        int invalidReportCount = parasoftFindingsBuildProcess.getInvalidReportCount();
+
+        // Then
+        Assertions.assertFalse(pmdReport.exists());
+        Assertions.assertFalse(pmdCpdReport.exists());
+        Assertions.assertEquals(status.name(), "FINISHED_FAILED");
+        Assertions.assertEquals(invalidReportCount, 1);
```

Comment:
It's better that if we can prevent the following printout when running "mvn test" by configure the exception as expected in the test:
![Uploading image.png…]()
@rwang-parasoft What do you think